internal/incident/restore_incidents_test.go

```diff
@@ -0,0 +1,223 @@
+// tags and name, and ensures that no error is returned, otherwise it will cause the entire test suite to fail.
+// Once the object has been successfully synchronised, an incident is created and synced with the database.
+func makeIncident(ctx context.Context, db *icingadb.DB, t *testing.T, recovered bool) *Incident {
+	ev := &event.Event{
+		Time:     time.Time{},
+		SourceId: 1,
+		Name:     makeRandomString(t),
+		Tags: map[string]string{ // Always generate unique object tags not to produce same object ID!
+			"host":    makeRandomString(t),
+			"service": makeRandomString(t),
+		},
+		ExtraTags: map[string]string{
+			"hostgroup/database-server": "",
+			"servicegroup/webserver":    "",
+		},
+	}
+
+	o, err := object.FromEvent(ctx, db, ev)
+	require.NoError(t, err)
+
+	i := NewIncident(db, o, &config.RuntimeConfig{}, nil)
+	i.StartedAt = types.UnixMilli(time.Now().Add(-2 * time.Hour).Truncate(time.Second))
+	i.Severity = event.SeverityCrit
+	if recovered {
+		i.Severity = event.SeverityOK
+		i.RecoveredAt = types.UnixMilli(time.Now())
+	}
+
+	tx, err := db.BeginTxx(ctx, nil)
+	require.NoError(t, err, "starting a transaction should not fail")
+	require.NoError(t, i.Sync(ctx, tx), "failed to insert incident")
+	require.NoError(t, tx.Commit(), "committing a transaction should not fail")
+
+	return i
+}
+
+func makeRandomString(t *testing.T) string {
+	buf := make([]byte, 20)
+	_, err := rand.Read(buf)
+	require.NoError(t, err, "failed to generate random string")
+
+	return fmt.Sprintf("%x", buf)
+}
+
+// getTestDB retrieves the database config from env variables, opens a new database and returns it.
+//
+// The test suite will be skipped if no environment variable is set, otherwise fails fatally when
+// invalid configurations are specified.
+func getTestDB(ctx context.Context, t *testing.T) *icingadb.DB {
+	c := &baseConfig.Database{}
+	require.NoError(t, defaults.Set(c), "applying config default should not fail")
+
+	if v, ok := os.LookupEnv("NOTIFICATIONS_TESTS_DB_TYPE"); ok {
+		c.Type = strings.ToLower(v)
+	} else {
+		t.Skipf("Environment %q not set, skipping test!", "NOTIFICATIONS_TESTS_DB_TYPE")
+	}
+
+	if v, ok := os.LookupEnv("NOTIFICATIONS_TESTS_DB"); ok {
+		c.Database = v
+	}
+	if v, ok := os.LookupEnv("NOTIFICATIONS_TESTS_DB_USER"); ok {
+		c.User = v
+	}
+	if v, ok := os.LookupEnv("NOTIFICATIONS_TESTS_DB_PASSWORD"); ok {
+		c.Password = v
+	}
+	if v, ok := os.LookupEnv("NOTIFICATIONS_TESTS_DB_HOST"); ok {
+		c.Host = v
+	}
+	if v, ok := os.LookupEnv("NOTIFICATIONS_TESTS_DB_PORT"); ok {
+		port, err := strconv.Atoi(v)
+		require.NoError(t, err, "invalid port provided")
+
+		c.Port = port
+	}
+
+	require.NoError(t, c.Validate(), "database config validation should not fail")
+
+	db, err := c.Open(logging.NewLogger(zaptest.NewLogger(t).Sugar(), time.Hour))
+	require.NoError(t, err, "connecting to database should not fail")
+	require.NoError(t, db.PingContext(ctx), "pinging the database should not fail")
+
+	return db
+}
```

Comment:
Please move this somewhere where it can be used by other test cases in the future, this will then become a useful part of #130.

After reading https://github.com/golang/go/issues/12120, the best way to do so seems to create a new package that's only imported from tests (and place it in a regular file, i.e. *not* `*_test.go`). I'd suggest `internal/testutils/`.